sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/OrcSourceSuite.scala

```diff
@@ -194,6 +196,24 @@ abstract class OrcSuite extends QueryTest with TestHiveSingleton with BeforeAndA
       Utils.deleteRecursively(location)
     }
   }
+
+  test("SPARK-21839: Add SQL config for ORC compression") {
+    val conf = sqlContext.sessionState.conf
+    assert(new OrcOptions(Map.empty[String, String], conf).compressionCodec == "SNAPPY")
+
+    // OrcOptions's parameters have a higher priority than SQL configuration.
+    withSQLConf(SQLConf.ORC_COMPRESSION.key -> "uncompressed") {
+      assert(new OrcOptions(Map.empty[String, String], conf).compressionCodec == "NONE")
+      assert(
+        new OrcOptions(Map("orc.compress" -> "zlib"), conf).compressionCodec == "ZLIB")
+    }
+
+    Seq("NONE", "SNAPPY", "ZLIB", "LZO").foreach { c =>
```

Comment:
// to test all the valid options of `spark.sql.orc.compression.codec`

> "none", "uncompressed", "snappy", "zlib", "lzo". 

You missed one of it.

Reply:
Ur, It's intentional. It's tested in the above. "UNCOMPRESSED" is replaced into "NONE"
So, I omitted here.

Reply:
To add that, we need `if` condition. I'll add that.